File: app/src/main/java/com/example/tetrisapp/util/FirebaseTokenUtil.java
package com.example.tetrisapp.util;

import com.example.tetrisapp.interfaces.TokenListener;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class FirebaseTokenUtil {
    public static void getFirebaseToken(TokenListener tokenListener){
        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
        if (user == null) return;

        user.getIdToken(true)
                .addOnCompleteListener(task -> {
                    if(task.isSuccessful())
                        tokenListener.getToken(task.getResult().getToken());
                });
    }
}

File: app/src/main/java/com/example/tetrisapp/util/FirebaseTokenUtil.kt
package com.example.tetrisapp.util

import com.google.android.gms.tasks.Task
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.GetTokenResult

object FirebaseTokenUtil {
    fun getFirebaseToken(tokenListener: (token: String?) -> Unit) {
        val user = FirebaseAuth.getInstance().currentUser
        if (user == null) {
            tokenListener(null)
            return
        }
        user.getIdToken(true)
            .addOnCompleteListener { task: Task<GetTokenResult> ->
                if (task.isSuccessful) {
                    tokenListener(task.result.token)
                } else {
                    tokenListener(null)
                }
            }
    }
}